### PR TITLE
Moar monster mods

### PR DESCRIPTION
**Pull request by MuteTiefling**
lots of new mischief for mobs, including more reactive gear. 

Disables drop chance for certain equipment items

prevents these equipment sets appearing on Spirit spawners.

### CODE REVIEW

File: kubejs/server_scripts/base/entity/spawned/equipment.js [hunk 1]
```diff
@@ -20,6 +20,10 @@ EntityEvents.spawned((event) => {
         return;
     }
 
+    // Don't apply buffs to Spirit spawner mobs
+    if (event.entity.fullNBT.hasOwnProperty('Corrupted')) {
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity/spawned/equipment.js:24]: Is this safe if the entity has no nbt?

File: kubejs/server_scripts/base/entity/spawned/equipment.js [hunk 2]
```diff
@@ -115,6 +149,8 @@ EntityEvents.spawned((event) => {
                     }
                 });
             }
+            console.log(mob_type);
+            console.log(event.entity.fullNBT);
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/entity/spawned/equipment.js:153]: This might get a bit spammy?

**Review by NielsPilgaard — APPROVED** [review]

File: kubejs/server_scripts/base/entity/spawned/equipment.js (hunk 1 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/base/entity/spawned/equipment.js:24]: They always have nbt of some sort.

File: kubejs/server_scripts/base/entity/spawned/equipment.js (hunk 2 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/base/entity/spawned/equipment.js:153]: Yep, meant to turn that off

Comment (MuteTiefling) [reply kubejs/server_scripts/base/entity/spawned/equipment.js]: logs disabled

**Pull request merged**